proxy/http2/Http2ConnectionState.cc

```diff
@@ -1186,8 +1186,13 @@ Http2ConnectionState::release_stream(Http2Stream *stream)
     if (total_client_streams_count == 0) {
       if (fini_received) {
         // We were shutting down, go ahead and terminate the session
+        // this is a member of Http2ConnectionState and will be freed
+        // when ua_session is destroyed
         ua_session->destroy();
-        ua_session = nullptr;
+
```

Comment:
so `destroy()` can free the memory of `Http2ConnectionState`?
If so, that's super dangerous. we might need to refactor shutting down logic as well. @shinrich @maskit @masaori335

Reply:
@zizhong Probably, it should be fine. I assume that `fini_received == true` & `total_client_streams_count == 0` guarantee that nobody needs the Http2ConnectionState instance anymore. But closing/releasing H2 session has been tricky, nobody can be sure.

Do you have any particular concerns?